src/main/scala/scalismo/mesh/PolyLine.scala

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2015 University of Basel, Graphics and Vision Research Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scalismo.mesh
+
+import scalismo.common.{ PointId, UnstructuredPointsDomain, Cell }
+import scalismo.geometry._
+
+import scala.util.Random
+
+/**
+ * Represents a line segment defined by two points.
+ */
+case class LineCell(ptId1: PointId, ptId2: PointId) extends Cell {
+
+  /** Identifiers of the points belonging to the cell*/
+  val pointIds = IndexedSeq(ptId1, ptId2)
+
+  /** Returns true if the given point identifier is part of the triangle cell*/
+  def containsPoint(ptId: PointId) = ptId1 == ptId || ptId2 == ptId
+
+  def toIndex2D = IntVector(ptId1.id, ptId2.id)
+}
+
+abstract class PolyLine[D <: Dim: NDSpace](val pointSet: UnstructuredPointsDomain[D], val topology: LineList) {
```

Comment:
also without walks, a consistent ordering (of all individual line segments) is still necessary if you want normals to point in a consistent direction (usually outside)